ReactNativeClient/lib/BaseApplication.js

```diff
@@ -529,6 +528,15 @@ class BaseApplication {
 				await FoldersScreenUtils.scheduleRefreshFolders();
 			}
 		}
+
+		if (action.type == 'SETTING_UPDATE_ONE' && action.key === 'preferredlightTheme' && !Setting.value('shouldUseDarkColors')) {
+			Setting.setValue('theme', action.value);
+		}
+
+		if (action.type == 'SETTING_UPDATE_ONE' && action.key === 'preferredDarkTheme' && Setting.value('shouldUseDarkColors')) {
+			Setting.setValue('theme', action.value);
+		}
```

Comment:
The logic above is not right as it's automatically changing the theme even when autodetect is off.

Reply:
Yes I intentionally didn't checked for AutoDetect option because these drop downs will only be visible when AutoDetect is ON hence one cannot change this settings from GUI without enabling AutoDetect option. If you still want to add the logic for AutoDetect option then I'll implement it.

Reply:
Yes the logic needs to be correct independently of the UI.